### ISSUE TITLE
member-ordering: doesn't check constructor

### ISSUE THREAD

**Issue by abierbaum** [palantir/tslint#1158]
### Bug Report 

- __TSLint version__: `3.8.0`
- __TypeScript version__: `1.8.10`
- __Running TSLint via__: CLI

#### TypeScript code being linted

with `tslint.json`:

```json
{
   "member-ordering": [
           true,
           "public-before-private",
           "static-before-instance",
           "variables-before-functions"
        ]
}
```

When checking this code:

```ts
class MyClass {
       foo() {}
       blah: number = 0;
}
```

I get the expected warning:

  > [30, 10]: Declaration of public instance member variable not allowed to appear after declaration of public instance member function

But when checking this code it passes. (renamed 'foo' to 'constructor')

```ts
class MyClass {
       constructor() {}
       blah: number = 0;
}
```

#### Expected behavior

I would expect the constructor case to have the same warning because there is a method before property definition.

**Comment by adidahiya** [comment]
Thanks for the bug report @abierbaum! Should be a relatively easy fix if anyone wants to send a PR.

**Comment by lowkay** [comment]
Hmm, now this includes checks against static methods too... I prefer constructors as the first thing after members and then methods after that. So that's an ordering that doesn't seem to be supported now, but it used to be accidentally because of this bug e.g.

```ts
class MyClass {
    private static staticMember: number = 7;

    private member: number;

    constructor(value) {
        this.member = value;
    }

    public static publicStaticMethod(): number { ... }

    private static privateStaticMethod(): number { ... }

    public publicMemberMethod(): number { ... }

    private privateMemberMethod(): number { ... }
}
```

Is there any plan to support this ordering style, which is a common C# style i.e. fields, constructors then methods while maintaining static/instance ordering within those sections?
